src/index.ts

```diff
@@ -0,0 +1,130 @@
+import * as path from "path"
+import * as fs from "@ts-common/fs"
+import * as md from "@ts-common/commonmark-to-markdown"
+import * as azureMd from "@azure/openapi-markdown"
+import * as ai from "@ts-common/async-iterator"
+import * as jp from "@ts-common/json-parser"
+import * as it from "@ts-common/iterator"
+import * as json from "@ts-common/json"
+import * as sm from "@ts-common/string-map"
+
+export const cli = async <T>(f: (path: string) => AsyncIterable<T>): Promise<number> => {
+  try {
+    const errors = await f("./")
+    let errorsNumber = 0
+    for await (const e of errors) {
+      console.error(e)
+      ++errorsNumber
+    }
+    console.log(`errors: ${errorsNumber}`)
+    return errorsNumber === 0 ? 0 : 1
+  } catch (e) {
+    console.error("INTERNAL ERROR")
+    console.error(e);
+    return 1
+  }
+}
+
+export type JsonParseError = {
+  readonly code: "JSON_PARSE",
+  readonly error: jp.ParseError
+}
+
+export type FileError = {
+  code: "NO_OPEN_API_FILE_FOUND" | "UNREFERENCED_OPEN_API_FILE"
+  readonly message: string
+  readonly readMeUrl: string
+  readonly openApiUrl: string
+}
+
+export type Error = JsonParseError | FileError
+
+export const avocado = (dir: string): ai.AsyncIterableEx<Error> =>
```

Comment:
we talked a couple of times about this, but i would recommend reducing as much as possible dependencies to ts-common utilties like AsyncIterable unless tehre is a **very good reason** (e.g.great perf savings). otherwise from what I look we get entagled in a series of custom calls and custom results which only the ts common libraries can accept ( i.e. som AysncIterableEx which it seems it is only accepted by some other ts-common library) which will make it difficult to move away in the future from.

Reply:
in general, i think we should take a depenency like that only if there is nothing out there that is popular and very well tested in the field against issues as opposed to our custom libraries that fits a blocking issue on our side

Reply:
I couldn't find well known/tested native TypeScript async iterator library.

Reply:
But, I hit some issues with using async iterator.